code/game/objects/effects/anomalies.dm

```diff
@@ -31,10 +31,13 @@
 
 
 /obj/effect/anomaly/proc/anomalyNeutralize()
-	new /obj/effect/effect/bad_smoke(loc)
+	var/turf/T = get_turf(src)
 
-	for(var/atom/movable/O in src)
-		O.loc = src.loc
+	new /obj/effect/effect/bad_smoke(T)
+
+	if(aSignal)
+		aSignal.forceMove(T)
+		aSignal = null
```

Comment:
With `aSignal` nulled out here, sticking `QDEL_NULL` back into `Destroy` might be wise, so an anomaly killed in a manner where it's not neutralized won't leave the beacon as a mess for something else to clean up